### PR TITLE
 BBG -[ inter-microservice-communicaiton]

### PR DESCRIPTION
**Pull request by dushansachinda**
The BBG guide written for building applications with a microservices architecture. In this article, we are considering a look at how the services within a system communicate with one another.

### CODE REVIEW

File: guide/passenger-management/passenger-management.bal [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// Initialize a JMS session on top of the created connection
+jms:Session jmsSession = new(conn, {
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+
+// Initialize a queue receiver using the created session
+endpoint jms:QueueReceiver jmsConsumer {
+    session:jmsSession,
+    queueName:"trip-passanger-notify"
```
Comment (afkham) [inline guide/passenger-management/passenger-management.bal]: s/passanger/passenger

Comment (dushansachinda) [reply guide/passenger-management/passenger-management.bal]: fixed!

File: guide/trip-management/trip-management.bal [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        // Order details
+        pickup.customerName = name.toString();
+        pickup.address = address.toString();
+        pickup.phonenumber = contact.toString();
+    
+        log:printInfo("Calling passenger management service:");
+      
+        // call passanger-management and get passagner orginization claims
+        json responseMessage;
+        http:Request passangermanagerReq;
```
Comment (Asitha) [inline guide/trip-management/trip-management.bal]: Shall we rename it to `passengerManagerReq`?

File: guide/trip-management/trip-management.bal [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        json pickupjson =  check <json>pickup;
+        passangermanagerReq.setJsonPayload(untaint pickupjson);
+        http:Response passangerResponse=  check passengerMgtEP -> post("/claims",passangermanagerReq);
```
Comment (Asitha) [inline guide/trip-management/trip-management.bal]: Shall we fix the formatting as 
```ballerina
... passangerResponse = check ...
```

File: guide/driver-management/driver-management.bal [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/jms;
+
+type Person record{
```
Comment (Asitha) [inline guide/driver-management/driver-management.bal]: Shall we fix code formatting
```ballerina
type Person record {

```

File: guide/dispatcher/dispatcher.bal [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    string tripID;
+    Driver driver;
+    Person person;
+    string time;
+};
+
+type Driver record {
+    string driverID;
+    string drivername;
+
```
Comment (VijithaEkanayake) [inline guide/dispatcher/dispatcher.bal]: Shall we remove this extra new line??

File: guide/dispatcher/dispatcher.bal [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        fork {
+            worker passengerNotification {
+                _ = jmsPassengerMgtNotifer -> send(queueMessage);
+            }
+            worker driverNotification {
+               _ = jmsDriverMgtNotifer -> send(queueMessage);
+            }
+
+        } join (all) (map collector) {
+    }
```
Comment (VijithaEkanayake) [inline guide/dispatcher/dispatcher.bal]: Minor formatting issue

File: guide/dispatcher/dispatcher.bal [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+
+// Initialize a queue receiver using the created session
+endpoint jms:QueueReceiver jmsConsumer {
+    session:jmsSession,
+    queueName:"trip-dispatcher"
+};
+
```
Comment (VijithaEkanayake) [inline guide/dispatcher/dispatcher.bal]: Do we need this much of new lines here??

File: guide/driver-management/driver-management.bal [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+
+// Initialize a queue receiver using the created session
+endpoint jms:QueueReceiver jmsConsumer {
+    session:jmsSession,
+    queueName:"trip-driver-notify"
+};
+
```
Comment (VijithaEkanayake) [inline guide/driver-management/driver-management.bal]: Do we need this much of new lines here?? I think having a new line would be enough between different constructs. WDYT?

**Pull request merged**